src/smtlib_reader.cpp

```diff
@@ -100,6 +119,13 @@ void SmtLibReader::set_logic(const string & logic)
 
         processed_logic =
             processed_logic.substr(len, processed_logic.length() - len);
+
+        // should also be in logic_sortkind_map
+        assert(logic_sortkind_map.find(sub) != logic_sortkind_map.end());
+        for (const SortKind sk : logic_sortkind_map.at(sub))
+        {
+          sortkinds_[smt::to_string(sk)] = sk;
```

Comment:
If we use UFs, the parser should still allow the declaration/definition of sorts named `Function`.

Reply:
Good point! Fixed that with fa2f52bb52fcbbe1a65543ce1ec06bb9e30be37e.